serialization/fontDataset.py

```diff
@@ -0,0 +1,89 @@
+import torch
+from torch.utils import data
+import font_pb2
+from os.path import join 
+import numpy as np
+import pandas as pd
+class Dataset(data.Dataset): 
+
+    def __init__(self, proto_dir, glyph_list, dimensions): 
+        '''
+        Parameters
+        ----------
+
+        proto_dir : directory of where all the glyph protos are located
+
+        glyph_list : file path that contains list of glyphs
+        '''
+        self.proto_dir = proto_dir
+        self.glyph_list = pd.read_csv(glyph_list,header=None)
+        self.dimensions = dimensions
+    def __len__(self):
+        return len(self.glyph_list)
+
+    def __getitem__(self,index): 
+        glyph = self.glyph_list.iloc[[index]]
+        bezier_points, contour_locations = readSerializedProto(glyph,self.proto_dir)
+        
+        # we can call a little function here to reshape and also add padding
+        # or alternativly it can be called on the other end
+        return (rollPadReshape(bezier_points,
+            contour_locations,self.dimensions), str(glyph))
```

Comment:
We should discuss this. I feel like it should return the plain data (perhaps unpadded), and we simply expose a utility function for users to call if they so choose.

Reply:
Lets talk about it on the call later today.